File: lib_base/src/main/java/cn/cbsd/base/event/RxBusImpl.java
package cn.cbsd.base.event;


import com.blankj.rxbus.RxBus;

/**
 * eventbus实现类，现在采取的是rxbus
 * @author zhenyanjun
 * @date 2019/1/8 10:56
 */

public class RxBusImpl implements IBus {

    private RxBusImpl() {
    }


    @Override
    public void register(Object object) {
    }

    @Override
    public void unregister(Object object) {
        RxBus.getDefault().unregister(object);
    }

    @Override
    public void post(BaseEvent event) {
        RxBus.getDefault().post(event);
    }

    @Override
    public void postSticky(BaseEvent event) {
        RxBus.getDefault().postSticky(event);
    }

    public <T extends BaseEvent> void subscribe(Object subscriber,
                                                RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribe(subscriber, callback);
    }

    public <T extends BaseEvent> void subscribeSticky(Object subscriber,
                                                      RxBus.Callback<T> callback) {
        RxBus.getDefault().subscribeSticky(subscriber, callback);
    }

    public static RxBusImpl get() {
        return Holder.instance;
    }

    private static class Holder {
        private static final RxBusImpl instance = new RxBusImpl();
    }
}

File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/event/RxBusImpl.kt
package cn.cbsd.mvplibrary.event


import com.blankj.rxbus.RxBus

/**
 * eventbus实现类
 * 现在采取的是rxbus
 * Created by wanglei on 2016/12/22.
 */

object RxBusImpl : IBus {

    override fun register(obj: Any) {}

    override fun unregister(obj: Any) {
        RxBus.getDefault().unregister(obj)
    }

    override fun post(event: AbsEvent) {
        RxBus.getDefault().post(event)
    }

    override fun postSticky(event: AbsEvent) {
        RxBus.getDefault().postSticky(event)
    }

    fun <T : AbsEvent> subscribe(subscriber: Any, callback: RxBus.Callback<T>) {
        RxBus.getDefault().subscribe(subscriber, callback)
    }

    fun <T : AbsEvent> subscribeSticky(subscriber: Any,callback: RxBus.Callback<T>) {
        RxBus.getDefault().subscribeSticky(subscriber, callback)
    }

}

